File: problem/PRG/불량사용자.swift
func solution(_ user_id:[String], _ banned_id:[String]) -> Int {
    var result: [[Int]] = Array(repeating: [], count: banned_id.count)
    for (index, banID) in banned_id.enumerated() {
        for (i, userID) in user_id.enumerated() {
            if userID.count != banID.count { continue }
            let starIndex = banID.enumerated().filter { $1 == "*" }.map { $0.offset }
            var tempBanID = banID.map { String($0) }
            let tempUserID = userID.map { String($0) }
            starIndex.forEach { tempBanID[$0] = tempUserID[$0] }
            if tempUserID.joined() == tempBanID.joined() {
                result[index].append(i)
            }
        }
    }
    var set = Set<[Int]>()
    func combinate(i: Int, select: [Int]) {
        if select.count == banned_id.count {
            set.update(with: select.sorted())
            return
        }
        var select = select
        let arr = result[i]
        for n in arr {
            if select.contains(n) { continue }
            select.append(n)
            combinate(i: i+1, select: select)
            select.removeLast()
        }
    }
    combinate(i: 0, select: [])
    return set.count
}

File: problem/PRG/불량사용자.py
def solution(user_id, banned_id):
    answer = 1
    all = []
    for i, ban in enumerate(banned_id):
        temp = []
        for user in user_id:
            if len(user) == len(ban):
                for j in range(len(ban)):
                    if user[j] == ban[j] or ban[j] == "*":
                        continue
                    else:
                        break
                else:
                    temp.append(user)
        all.append(temp)

    cases = []

    def dfs(i, select):
        if len(select) == len(banned_id):
            if sorted(select) not in cases:
                cases.append(sorted(select))
            return

        arr = all[i]
        for a in arr:
            if a in select:
                continue
            select.append(a)
            dfs(i + 1, select)
            select.pop()

    dfs(0, [])
    return len(cases)
